zerver/views/registration.py

```diff
@@ -345,14 +345,19 @@ def create_realm(request: HttpRequest, creation_key: Optional[Text]=None) -> Htt
         if form.is_valid():
             email = form.cleaned_data['email']
             activation_url = prepare_activation_url(email, request, realm_creation=True)
+            if have_valid_key:
+                # The user has a token created from the server command line;
+                # skip confirming the email is theirs, taking their word for it.
+                # This is essential on first install if the admin hasn't stopped
+                # to configure outbound email up front, or it isn't working yet.
+                RealmCreationKey.objects.get(creation_key=creation_key).delete()
+                return HttpResponseRedirect(activation_url)
+
```

Comment:
I haven't thought through whether this implementation creates such a problem, but I think this might create a gap where if we're doing `generate_realm_creation_link` on a multi-tenant server (e.g. via the old as a zulipchat.com beta form), the server admin isn't the person who is creating the realm, and we'd have no email validation there.  Probably the right solution is to have a an additional boolean on the `RealmCreationKey` object for whether email is assumed validated; make the default for the management command that it is assumed validated (and default off for the beta form use case).

Reply:
Hmm, reasonable.

Just pushed an updated version that adds this boolean; PTAL. (Also merged the initial few commits that were before this one.)